tests/library/browsercontext-fetch-algorithms.spec.ts

```diff
@@ -0,0 +1,98 @@
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+import { contextTest as it, expect } from '../config/browserTest';
+import util from 'util';
+import zlib from 'zlib';
+
+const gzip = util.promisify(zlib.gzip);
+const deflate = util.promisify(zlib.deflate);
+const brotliCompress = util.promisify(zlib.brotliCompress);
+
+it.skip(({ mode }) => mode !== 'default');
+
+it.describe('algorithms', () => {
+  const responseBody = 'str';
+
+  for (const [type, zipped] of Object.entries({
+    gzip: gzip(responseBody),
+    deflate: deflate(responseBody),
+    br: brotliCompress(responseBody)
+  })) {
+    it.describe(`${type} decompression`, () => {
+      it(`should support decompression`, async ({ context, server }) => {
+        server.setRoute('/compressed', async (req, res) => {
+          res.setHeader('Content-Encoding', type);
+          res.end(await zipped);
+        });
+
+        const response = await context.request.get(server.PREFIX + '/compressed');
+        expect(await response.text()).toEqual(responseBody);
+      });
+
+      it(`should not fail if response content-length header is missing (${type})`, async ({ context, server }) => {
+        server.setRoute('/compressed', async (req, res) => {
+          res.setHeader('Content-Encoding', type);
+          res.removeHeader('Content-Length');
+          res.end(await zipped);
+        });
+
+        const response = await context.request.get(server.PREFIX + '/compressed');
+        expect(await response.text()).toEqual(responseBody);
+      });
+
+      it('should not fail with chunked responses (without Content-Length header)', async ({ context, server }) => {
+        server.setRoute('/compressed', async (req, res) => {
+          res.setHeader('Content-Encoding', type);
+          res.setHeader('Transfer-Encoding', 'chunked');
+          res.removeHeader('Content-Length');
+          res.write(await zipped);
+          res.end();
+        });
+
+        const response = await context.request.get(server.PREFIX + '/compressed');
+        expect(await response.text()).toEqual(responseBody);
+      });
+
+      it('should not fail with an empty response without content-length header (Z_BUF_ERROR)', async ({ context, server }) => {
+        server.setRoute('/compressed', async (req, res) => {
+          res.setHeader('Content-Encoding', type);
+          res.removeHeader('Content-Length');
+          res.end();
+        });
+
+        const response = await context.request.get(server.PREFIX + '/compressed');
+        expect(await response.text()).toEqual('');
+      });
+
+      it('should not fail with an empty response with content-length header (Z_BUF_ERROR)', async ({ context, server }) => {
+        server.setRoute('/compressed', async (req, res) => {
+          res.setHeader('Content-Encoding', type);
```

Comment:
I believe we need to set Content-length manually here.

Reply:
These are checking if things work w/o content length...